src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -185,6 +179,49 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite {
     }
   }
 
+  test("Verify the location of injected shuffle for Hybrid Scan.") {
+    val dataPath = systemPath.toString + "/hbtable"
+    import spark.implicits._
+    Seq((1, "name1", 12), (2, "name2", 10))
+      .toDF("id", "name", "age")
+      .write
+      .mode("overwrite")
+      .parquet(dataPath)
+
+    val df = spark.read.parquet(dataPath)
+    val query = df.filter(df("id") >= 3).select("id", "name")
+    val bucketSpec = BucketSpec(100, Seq("id"), Seq())
+    val shuffled = RuleUtils.transformPlanToShuffleUsingIndexSpec(
+      bucketSpec,
+      query.queryExecution.optimizedPlan)
+
+    assert((shuffled collect {
+      case RepartitionByExpression(attrs: Seq[Expression], p: Project, numBuckets: Int) =>
+        assert(numBuckets == 100)
+        assert(attrs.size == 1)
+        assert(attrs.head.asInstanceOf[Attribute].name.contains("id"))
+        assert(p.projectList.exists(a => a.name.equals("id")) && p.projectList.exists(a =>
+          a.name.equals("name")))
+        true
+    }).length == 1)
+
+    val bucketSpec2 = BucketSpec(100, Seq("age"), Seq())
```

Comment:
Can you make a short comment on what you are testing for each scenario?

Reply:
Done